main/test/ca/uwaterloo/flix/library/TestMutList.flix

```diff
@@ -933,22 +947,21 @@ namespace TestMutList {
     @test
     def shrinkTo!01(): Bool & Impure =
         let v = MutList.new();
-        let c = MutList.capacity(v);
         MutList.shrinkTo!(0, v);
-        c == MutList.capacity(v) and MutList.length(v) == 0
+        MutList.length(v) == 0
 
     @test
```

Comment:
What do you guys think?

Reply:
I disagree. Part of the contract of `shrink!` is that we reduce the capacity. If we don't test that, then `shrink!` could just be an `id` function, for example.

If we want to maintain a high-level view of this data structure, maybe we should consider whether these functions that theoretically have no observable effect really belong in the library.

Reply:
I agree with @mlutze.
If we view the test suite as an operational specification of the data structure then it makes sense to test the implementation of functions which only perform side-effects. The user shouldn't observe the effects directly, but the test suite should have more privileges in that regard, since it's the only way for us to verify that it lives up to said specification. They do test implementation, but in my opinion that can be slightly remedied by testing the desired relative change, and **not** testing constants (i.e. we should prefer `capacity(v) == 2*c` rather than `capacity(v) == 16` as an example). Additionally, if we go with this option we should maybe consider splitting tests into interface tests and implementation tests, although, this is mostly a readability issue.

To add to @mlutze's second point: If it's important for the user to manipulate the internal array, why don't they just use an array instead? The array in MutList is very much an implementation detail and should not be observable by the user.